src/crimson/admin/osd_admin.cc

```diff
@@ -230,4 +231,35 @@ class AssertAlwaysHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<AssertAlwaysHook>();
 
+/**
+ * A Seastar admin hook: fetching the values of configured metrics
+ */
+class SeastarMetricsHook : public AdminSocketHook {
+public:
+  SeastarMetricsHook()  :
+    AdminSocketHook("perf dump_seastar",
+		   "perf dump_seastar",
+		   "dump current configured seastar metrics and their values")
+  {}
+  seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+				      std::string_view format,
+				      ceph::bufferlist&& input) const final
+  {
+    unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+    f->open_object_section("perf_dump_seastar");
+    auto value_map = seastar::metrics::impl::get_value_map();
```

Comment:
The method signature is: `const value_map& get_value_map() const`.

Is the `value_map` copy intentional? If not, then maybe `for (const auto& mf : seastar::metrics::impl::get_value_map())` would do? Otherwise a comment emphasizing the intention seems justified.